### PR TITLE
Add table oci_core_volume_backup_policy. CLoses #96

### LINKED ISSUE #96: Add table oci_core_volume_backup_policy
**References**
https://github.com/oracle/oci-go-sdk/blob/master/core/volume_backup_policy.go

### PR DESCRIPTION
**Pull request by rajeshbal65**
# Integration test logs
<details>

  
```
Test names: [ 'tests/oci_core_volume_backup_policy' ]
No env file present for the current environment:  staging
 Falling back to .env config
No env file present for the current environment:  staging

SETUP: tests/oci_core_volume_backup_policy []

PRETEST: tests/oci_core_volume_backup_policy

TEST: tests/oci_core_volume_backup_policy
Running terraform
oci_core_volume_backup_policy.named_test_resource: Creating...
oci_core_volume_backup_policy.named_test_resource: Creation complete after 2s [id=ocid1.volumebackuppolicy.oc1.ap-mumbai-1.aaaaaaaa5rfkicmz6by7bxwsvhbd5cglya7acbiss3lfay4tf3zii5lmqu6a]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

Outputs:

region = "ap-mumbai-1"
resource_id = "ocid1.volumebackuppolicy.oc1.ap-mumbai-1.aaaaaaaa5rfkicmz6by7bxwsvhbd5cglya7acbiss3lfay4tf3zii5lmqu6a"
resource_name = "steampipetest1616"
tenancy_ocid = "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq"

Running SQL query: test-get-query.sql

[
  {
    "display_name": "steampipetest1616",
    "freeform_tags": {
      "Name": "steampipetest1616"
    },
    "id": "ocid1.volumebackuppolicy.oc1.ap-mumbai-1.aaaaaaaa5rfkicmz6by7bxwsvhbd5cglya7acbiss3lfay4tf3zii5lmqu6a",
    "region": "ap-mumbai-1"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql

[
  {
    "display_name": "steampipetest1616",
    "freeform_tags": {
      "Name": "steampipetest1616"
    },
    "id": "ocid1.volumebackuppolicy.oc1.ap-mumbai-1.aaaaaaaa5rfkicmz6by7bxwsvhbd5cglya7acbiss3lfay4tf3zii5lmqu6a",
    "region": "ap-mumbai-1"
  }
]
✔ PASSED

Running SQL query: test-not-found-query.sql

null
✔ PASSED

Running SQL query: test-turbot-query.sql

[
  {
    "tenant_id": "ocid1.tenancy.oc1..aaaaaaaahnm7gleh5soecxzjetci3yjjnjqmfkr4po3hoz4p4h2q37cyljaq",
    "title": "steampipetest1616"
  }
]
✔ PASSED

POSTTEST: tests/oci_core_volume_backup_policy

TEARDOWN: tests/oci_core_volume_backup_policy

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
### Basic info

```sql
select
  id,
  display_name,
  time_created,
  region,
  tags
from
  oci_core_volume_backup_policy;

+-------------------------------------------------------------------------------------------------------+--------------+---------------------+-------------+------------------------------------------------------------------------------+
| id                                                                                                    | display_name | time_created        | region      | tags                                                                         |
+-------------------------------------------------------------------------------------------------------+--------------+---------------------+-------------+------------------------------------------------------------------------------+
| ocid1.volumebackuppolicy.oc1.ap-mumbai-1.aaaaaaaah5yqtcvtvufi567gdjcsgrwg4nskuflfn6j2rtznnv2pgz2q2pta | test2        | 2021-04-19 13:43:37 | ap-mumbai-1 | {"CreatedBy":"rajesh_bal","CreatedOn":"2021-04-19T13:43:37.523Z"}            |
| ocid1.volumebackuppolicy.oc1.ap-mumbai-1.aaaaaaaahtfydyy3vbbaexrhggur6p2l2qhdz4halvsno7z7n6kjjtb7ui6a | test         | 2021-04-19 12:28:33 | ap-mumbai-1 | {"CreatedBy":"rajesh_bal","CreatedOn":"2021-04-19T12:28:33.182Z","df":"wfw"} |
+-------------------------------------------------------------------------------------------------------+--------------+---------------------+-------------+------------------------------------------------------------------------------+
```

### List the destination regions of volume back up policy

```sql
select
  id,
  display_name,
  destination_region
from
  oci_core_volume_backup_policy;

+-------------------------------------------------------------------------------------------------------+--------------+--------------------+
| id                                                                                                    | display_name | destination_region |
+-------------------------------------------------------------------------------------------------------+--------------+--------------------+
| ocid1.volumebackuppolicy.oc1.ap-mumbai-1.aaaaaaaah5yqtcvtvufi567gdjcsgrwg4nskuflfn6j2rtznnv2pgz2q2pta | test2        | ap-hyderabad-1     |
| ocid1.volumebackuppolicy.oc1.ap-mumbai-1.aaaaaaaahtfydyy3vbbaexrhggur6p2l2qhdz4halvsno7z7n6kjjtb7ui6a | test         | ap-hyderabad-1     |
+-------------------------------------------------------------------------------------------------------+--------------+--------------------+
```

### List schedule info of volume backup policy

```sql
select
  id,
  display_name,
  s ->> 'backupType' as backup_type,
  s ->> 'dayOfMonth' as day_of_month,
  s ->> 'dayOfWeek' as day_of_week,
  s ->> 'hourOfDay' as hour_of_day,
  s ->> 'month' as month,
  s ->> 'offsetSeconds' as offset_econds,
  s ->> 'offsetType' as offset_type,
  s ->> 'period' as offset_econds,
  s ->> 'retentionSeconds' as retention_seconds,
  s ->> 'timeZone' as time_zone
from
  oci_core_volume_backup_policy,
  jsonb_array_elements(schedules) as s;

+-------------------------------------------------------------------------------------------------------+--------------+-------------+--------------+-------------+-------------+---------+---------------+-------------+---------------+-------------------+-----------+
| id                                                                                                    | display_name | backup_type | day_of_month | day_of_week | hour_of_day | month   | offset_econds | offset_type | offset_econds | retention_seconds | time_zone |
+-------------------------------------------------------------------------------------------------------+--------------+-------------+--------------+-------------+-------------+---------+---------------+-------------+---------------+-------------------+-----------+
| ocid1.volumebackuppolicy.oc1.ap-mumbai-1.aaaaaaaahtfydyy3vbbaexrhggur6p2l2qhdz4halvsno7z7n6kjjtb7ui6a | test         | INCREMENTAL | 1            | MONDAY      | 0           | JANUARY | 0             | STRUCTURED  | ONE_WEEK      | 604800            | UTC       |
| ocid1.volumebackuppolicy.oc1.ap-mumbai-1.aaaaaaaahtfydyy3vbbaexrhggur6p2l2qhdz4halvsno7z7n6kjjtb7ui6a | test         | FULL        | 1            | TUESDAY     | 0           | JANUARY | 0             | STRUCTURED  | ONE_YEAR      | 31536000          | UTC       |
+-------------------------------------------------------------------------------------------------------+--------------+-------------+--------------+-------------+-------------+---------+---------------+-------------+---------------+-------------------+-----------+
```

### List full backup types volume backup policy

```sql
select
  id,
  display_name,
  s ->> 'backupType' as backup_type
from
  oci_core_volume_backup_policy,
  jsonb_array_elements(schedules) as s
where
  s ->> 'backupType' = 'FULL';

+-------------------------------------------------------------------------------------------------------+--------------+-------------+
| id                                                                                                    | display_name | backup_type |
+-------------------------------------------------------------------------------------------------------+--------------+-------------+
| ocid1.volumebackuppolicy.oc1.ap-mumbai-1.aaaaaaaahtfydyy3vbbaexrhggur6p2l2qhdz4halvsno7z7n6kjjtb7ui6a | test         | FULL        |
+-------------------------------------------------------------------------------------------------------+--------------+-------------+
```
```
</details>

### CODE REVIEW

File: oci/table_oci_core_volume_backup_policy.go [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+	logger.Debug("core.getCoreVolumeBackupPolicy", "Compartment", compartment, "OCI_REGION", region)
+
+	// Restrict the api call to only root compartment/ per region
+	if !strings.HasPrefix(compartment, "ocid1.tenancy.oc1") {
+		return nil, nil
+	}
+
+	id := d.KeyColumnQuals["id"].GetStringValue()
+
+	// handle empty volume id in get call
```
Comment (khushboo9024) [inline oci/table_oci_core_volume_backup_policy.go:170]:
```suggestion
	// handle empty volume backup policy id in get call
```

File: oci/table_oci_core_volume_backup_policy.go [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+			// Standard OCI columns
+			{
+				Name:        "region",
+				Description: ColumnDescriptionRegion,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Id").Transform(ociRegionName),
+			},
+			{
+				Name:        "compartment_id",
+				Description: "ColumnDescriptionCompartment",
```
Comment (khushboo9024) [inline oci/table_oci_core_volume_backup_policy.go:98]:
```suggestion
				Description: ColumnDescriptionCompartment,
```

**Review by khushboo9024 — CHANGES_REQUESTED** [review]
Minor changes required.

File: docs/tables/oci_core_volume_backup_policy.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  display_name,
+  time_created,
+  region,
+  tags
+from
+  oci_core_volume_backup_policy;
+```
+
+
+### List the destination regions of volume back up policy
```
Comment (khushboo9024) [inline docs/tables/oci_core_volume_backup_policy.md:21]:
```suggestion
### List the destination regions of volume backup policy
```

File: oci/table_oci_core_volume_backup_policy.go [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+)
+
+//// TABLE DEFINITION
+
+func tableCoreVolumeBackupPolicy(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "oci_core_volume_backup_policy",
+		Description: "OCI Core Volume Backup Policy",
+		Get: &plugin.GetConfig{
+			KeyColumns: plugin.AnyColumn([]string{"id"}),
```
Comment (khushboo9024) [inline oci/table_oci_core_volume_backup_policy.go:22]:
```suggestion
			KeyColumns: plugin.SingleColumn("id"),
```

**Review by khushboo9024 — CHANGES_REQUESTED** [review]
Please make suggested changes and resolve merge conflict.

**Review by khushboo9024 — APPROVED** [review]
LGTM

**Review by rajmohanty17 — APPROVED** [review]
LGTM

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**